js/task_with_timeout.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * vim: ts=4:sw=4:expandtab
+ */
+(function () {
+    window.Whisper = window.Whisper || {};
+
+    Whisper.createTaskWithTimeout = function(task, id) {
```

Comment:
This is entirely designed to reduce the impact of hangs, so I'd be fine putting the default timeout at, 30 seconds, a minute, or even more if you think legitimate scenarios will take that long.

For debugging scenarios sensitive to this, I suspect that a quick change to the default timeout might be enough of a workaround. I've run into difficulties due to the time travel detection stuff while debugging, so that approach has worked for me.